eth/stagedsync/stage_headers_new.go

```diff
@@ -261,6 +261,7 @@ func fixCanonicalChain(logPrefix string, height uint64, hash common.Hash, tx eth
 		ancestor := rawdb.ReadHeader(tx, ancestorHash, ancestorHeight)
 		if ancestor == nil {
 			log.Error("ancestor nil", "height", ancestorHeight, "hash", ancestorHash)
```

Comment:
Is this getting triggered during the cancellation?

Reply:
no

Reply:
````
INFO [05-20|10:57:40.689] Initialising Ethereum protocol           network=1
DEBUG[05-20|10:57:42.189] TCP listener up                          addr=[::]:30303
DEBUG[05-20|10:57:42.189] UDP listener up                          addr=[::]:30303
INFO [05-20|10:57:42.214] New local node record                    seq=18 id=a0cc79e4c94e0bb9 ip=127.0.0.1 udp=30303 tcp=30303
ERROR[05-20|10:57:42.215] ancestor nil                             height=245760 hash="583a8c…be886a"
INFO [05-20|10:57:42.215] New local node record                    seq=19 id=a0cc79e4c94e0bb9 ip=127.0.0.1 udp=30303 tcp=30303
INFO [05-20|10:57:42.215] Started P2P networking                   self=enode://a8fc622ea544177f86172fd431929a38964f675c55e0347fcf47b565bb0a9af69768a8259bcb0682b564e9bfa14b860af4a479c9d1ad4190b20b8ccd1119d2b2@127.0.0.1:30303
DEBUG[05-20|10:57:42.670] [2/14 BlockHashes] Extraction finished   it took=454.492531ms
DEBUG[05-20|10:57:43.045] [2/14 BlockHashes] Committed batch       bucket=H records=245760 current key=final alloc=331.25MiB sys=478.24MiB numGC=25
DEBUG[05-20|10:57:43.045] [2/14 BlockHashes] Collection finished   it took=375.125957ms
INFO [05-20|10:57:43.096] [3/14 Bodies] Processing bodies...       from=0 to=245760
DEBUG[05-20|10:57:43.320] Adding p2p peer                          peercount=1 id=bd741aab3d60f30d conn=dyndial addr=167.172.135.221:30303 name=Geth/v1.10.1-stable/...
DEBUG[05-20|10:57:43.320] [bd741aab3d60f30da611c07d3378113f0553140895022300857673ccb6c28975] Start with peer
DEBUG[05-20|10:57:43.340] Adding p2p peer                          peercount=2 id=8d1da4c5bc682f8f conn=dyndial addr=52.231.165.108:30303  name=Geth/v1.10.2-unstabl...
DEBUG[05-20|10:57:43.340] [8d1da4c5bc682f8f8ad141f60d2889fa4232144c7d2b29b88f81ca33e586ef9d] Start with peer
DEBUG[05-20|10:57:43.517] [8d1da4c5bc682f8f8ad141f60d2889fa4232144c7d2b29b88f81ca33e586ef9d] Received status message OK name=Geth/v1.10.2-unstabl...
DEBUG[05-20|10:57:43.522] [8d1da4c5bc682f8f8ad141f60d2889fa4232144c7d2b29b88f81ca33e586ef9d] Error while running peer: reading message: EOF
DEBUG[05-20|10:57:43.522] Removing p2p peer                        peercount=1 id=8d1da4c5bc682f8f duration=182.218ms req=true err="too many peers"
DEBUG[05-20|10:57:43.617] [bd741aab3d60f30da611c07d3378113f0553140895022300857673ccb6c28975] Received status message OK name=Geth/v1.10.1-stable/...
```

Reply:
Maybe it's result of previous cancelation - but cancelation must leave db in consistent state